go/vt/vttablet/tabletserver/vstreamer/rowstreamer.go

```diff
@@ -200,6 +200,8 @@ func (rs *rowStreamer) buildSelect() (string, error) {
 		buf.Myprintf("%s%v", prefix, sqlparser.NewColIdent(rs.plan.Table.Fields[pk].Name))
 		prefix = ", "
 	}
+
+	buf.Myprintf("LIMIT 1000000")
```

Comment:
think you need to add a space before the `LIMIT` here since i don't think the buffer has a space at the end of it.